src/main/interface/tramp_protocol.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "drivers/serial.h"
+
+#define TRAMP_SERIAL_OPTIONS            SERIAL_NOT_INVERTED | SERIAL_BIDIR
+#define TRAMP_BAUD                      9600
+#define TRAMP_PAYLOAD_LENGTH            12
+
+typedef struct trampSettings_s {
+    uint16_t frequency;
+    uint16_t power;
+    uint8_t raceModeEnabled;
+    uint8_t pitModeEnabled;
+} trampSettings_t;
+
+typedef struct trampFrameHeader_s {
+    uint8_t syncStart;
+    uint8_t command;
+} trampFrameHeader_t;
+
+typedef struct trampFrameFooter_s {
+    uint8_t crc;
+    uint8_t syncStop;
+} trampFrameFooter_t;
+
+typedef union trampPayload_u {
+    uint8_t buf[TRAMP_PAYLOAD_LENGTH];
+    trampSettings_t settings;
+    uint16_t frequency;
+    uint16_t power;
+    uint8_t active;
+} trampPayload_t;
+
+typedef struct trampFrame_s {
+    trampFrameHeader_t header;
+    trampPayload_t payload;
+    trampFrameFooter_t footer;
+} trampFrame_t;
```

Comment:
To be safe, `__attribute__((packed))` should/could be added.

Reply:
Yes, need to revise size and padding to ensure they actually are packed as expected

Reply:
Also, when dealing with externals specs, and no external headers are available to include, you could also make absolutely sure that all struct defs results in the correct sizes, as specified externally. Use for example this:
`STATIC_ASSERT( sizeof(trampFrame_t) == TRAMP_FRAME_LENGHT trampFrame_t);`

Reply:
Yes, if you have a struct defined by yourself, the only static assert which has a right to live would be `sizeof(header) == 2`, to ensure you didn't mess the alignment and padding. :)

Reply:
I'll pack all structs to remove the possibility of unwanted padding.